perf/kernbench.py

```diff
@@ -0,0 +1,145 @@
+        if not os.path.isfile('vmlinux'):
+            self.error("No vmlinux found, kernel build failed")
+
+    def setUp(self):
+        """
+        setting up the env for the kernel building
+        """
+        smg = SoftwareManager()
+        deps = ['gcc', 'make', 'automake', 'autoconf', 'time']
+        for package in deps:
+            if smg.check_installed(package) and not smg.install(package):
+                self.error(package + ' is needed for the test to be run')
+
+    def to_seconds(self, time_string):
+        """Converts a string in M+:SS.SS format to S+.SS"""
+        elts = time_string.split(':')
+        if len(elts) == 1:
+            return time_string
+        return str(int(elts[0]) * 60 + float(elts[1]))
+
+    def extract_all_time_results(self, results_string):
+        """Extract user, system, and elapsed times into a list of tuples"""
+        pattern = re.compile(r"(.*?)user (.*?)system (.*?)elapsed")
+        results = []
+        for result in pattern.findall(results_string):
+            results.append(tuple([self.to_seconds(elt) for elt in result]))
+        return results
+
+    def test(self):
+        """
+        Kernel build Test
+        """
+        config = self.params.get('config_file', default='config')
+        config_path = os.path.join(self.datadir, config)
+        iterations = self.params.get('runs', default=1)
+        threads = self.params.get('cpus', default=None)
+        version = self.params.get('version', default='4.6')
+        if iterations is None:
+            # Setting the default iteration as one
+            iterations = 1
+        if version is None:
+            # setting the default version 4.6
+            version = '4.6'
+        if threads is None:
+            # We will use 2 workers of each type for each CPU detected
+            threads = 2 * multiprocessing.cpu_count()
+        if config is None:
+            # Setting default config file as the 'config' in data Directory
+            config_path = os.path.join(self.datadir, config)
+        # initialize kernel object
+        kern = kernel.KernelBuild(
+            version, config_path=config_path, work_dir=None, data_dirs=None)
+        # Seeting the default kernel URL
+        kern.URL = 'https://www.kernel.org/pub/linux/kernel/v4.x/'
+        # download the kernel tarball from kernel.org
+        kern.download()
+        # Uncompress the kernel archive to the work directory
+        kern.uncompress()
+        # Running configure script
+        kern.configure()
+        log.info("Starting build the kernel")
+        timefile = ("%s/time_file" % kern.build_dir)
+        # Build kernel
+        user_time = 0
+        system_time = 0
+        elapsed_time = 0
+        for run in range(iterations):
+            log.info("Iteration: %s" % run)
+            self.time_build(kern, threads, timefile, "")
+            # Processing the timefile
+            results = open(timefile).readline()
```

Comment:
can `results = open(timefile).readline()` be` results = open(timefile).readline().strip()`  ?